### PR TITLE
Issue-2807 - added some inefficient casts and some enhanced casts (ma…

### PR DESCRIPTION
**Pull request by leozhekov**
Issue-2807 - added some inefficient casts and some enhanced casts (mainly on SV) to be highlighted on the timeline and updated all statistics of skills, talents and traits to use the new statistics module.

### CODE REVIEW

**Comment by leozhekov** [comment]
An album with screenshots of the highlights: https://imgur.com/a/OTcvV8c

**Comment by Sharrq** [comment]
Did some unrelated changes get committed with this? There are a LOT of changes in here that dont seem to be timeline highlights.

In general, its a good idea to split things into multiple PRs instead of one massive one. There are 79 files changed in this PR and its going to take a while to sort through them all.

**Comment by leozhekov** [comment]
> Did some unrelated changes get committed with this? There are a LOT of changes in here that dont seem to be timeline highlights.
> 
> In general, its a good idea to split things into multiple PRs instead of one massive one. There are 79 files changed in this PR and its going to take a while to sort through them all.

While I agree with this statement, Putro and I decided that a higher priority would be updating the modules with the new statistics. Since there are few highlights made, they are included in the changes.

**Comment by Sharrq** [comment]
I'm not saying updating the statistics isn't important, Im saying that putting the statistics updates for all three specs plus the timeline highlights into one massive PR makes it very difficult to review. Because we need to go through 79 files to make sure all of them are good and it's going to take a long time.

For the future, this could have been like 4 PRs (one for the timeline stuff, one for the survival statistics, one for mm statistics, one for bm statistics) which would make it a lot easier to review.

So just expect this to take a long time to get approved since its going to be a pain in the ass to review it all

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -38,4 +38,4 @@
       <option name="m_ignoreAnnotatedVariables" value="false" />
     </inspection_tool>
   </profile>
-</component>
```
[the remark this answers is not in the document]
Comment (leozhekov) [reply .idea/inspectionProfiles/Project_Default.xml]: I don't know. .idea should be ignored.

File: src/parser/hunter/survival/modules/spells/RaptorStrike.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * A vicious slash dealing (70% of Attack power) Physical damage.
+ *
+ * Example log: https://www.warcraftlogs.com/reports/BjmyHd9zt8RYJrWA/#fight=3&source=1
+ */
+
+class RaptorStrike extends Analyzer {
+
+  constructor(...args) {
+    super(...args);
+    this.active = !this.selectedCombatant.hasTalent(SPELLS.MONGOOSE_BITE_TALENT.id);
```
[the remark this answers is not in the document]
Comment (leozhekov) [reply src/parser/hunter/survival/modules/spells/RaptorStrike.js]: I'll add it in as soon as I have time.

**Comment by leozhekov** [comment]
> Had to glance quickly through a lot of the updated statisticboxes to get through this, so i'm trusting you made those right. I did notice that you generally had `STATISTIC_ORDER` imported from various different sources, maybe since you're already updating all of them you can keep it consistent and import from `interface/others/STATISTIC_ORDER` like you do in one of them.

I'll look into it!

**Pull request merged**